sklearn/utils/tests/test_stats.py

```diff
@@ -12,121 +12,151 @@
 from sklearn.utils._array_api import device as array_device
 from sklearn.utils.estimator_checks import _array_api_for_tests
 from sklearn.utils.fixes import np_version, parse_version
-from sklearn.utils.stats import _averaged_weighted_percentile, _weighted_percentile
+from sklearn.utils.stats import _weighted_percentile
 
 
-def test_averaged_weighted_median():
-    y = np.array([0, 1, 2, 3, 4, 5])
-    sw = np.array([1, 1, 1, 1, 1, 1])
+@pytest.mark.parametrize("average", [True, False])
+@pytest.mark.parametrize("size", [10, 15])
+def test_weighted_percentile_matches_median(size, average):
+    """Ensure `_weighted_percentile` matches `median` when expected.
 
-    score = _averaged_weighted_percentile(y, sw, 50)
+    With unit `sample_weight`, `_weighted_percentile` should match median except
+    when `average=False` and the number of samples is even.
+    For an even array and `average=False`, `percentile_rank=50` gives the lower
+    of the two 'middle' values, that would have been averaged for `median` calculation.
+    """
+    y = np.arange(size)
+    sample_weight = np.ones_like(y)
 
-    assert score == np.median(y)
+    score = _weighted_percentile(y, sample_weight, 50, average=average)
 
+    # `_weighted_percentile(average=False)` does not match `median` when n is even
+    if size == 10 and average is False:
```

Comment:
This makes the code both more explicit and still correct if we ever change the parametrization of `size`. 

```suggestion
    if size % 2 == 0 and average is False:
```